Scripts/RemediationScripts/Remediate-SetClusterProtectionLevelForServiceFabric.ps1

```diff
@@ -0,0 +1,731 @@
+    $backupFolderPath = "$([Environment]::GetFolderPath('LocalApplicationData'))\AzTS\Remediation\Subscriptions\$($context.Subscription.SubscriptionId.replace('-','_'))\$($(Get-Date).ToString('yyyyMMddhhmm'))\SetServiceFabricWithClusterProtectionLevel"
+
+    if (-not (Test-Path -Path $backupFolderPath))
+    {
+        New-Item -ItemType Directory -Path $backupFolderPath | Out-Null
+    }
+ 
+    Write-Host $([Constants]::DoubleDashLine)
+    Write-Host "[Step 3 of 4] Backing up Service Fabric(s) details..."
+    Write-Host $([Constants]::SingleDashLine)
+
+    if ([String]::IsNullOrWhiteSpace($FilePath))
+    {
+        
+    
+        # Backing up Service Fabric(s) details.
+        $backupFile = "$($backupFolderPath)\ServiceFabricDetailsBackUp.csv"
+
+        $ServiceFabricWithoutEncryptandSign | Export-CSV -Path $backupFile -NoTypeInformation
+
+        Write-Host "Service Fabric(s) details have been backed up to" -NoNewline
+        Write-Host " [$($backupFile)]" -ForegroundColor $([Constants]::MessageType.Update)
+    }
+    else
+    {
+        Write-Host "Skipped as -FilePath is provided" -ForegroundColor $([Constants]::MessageType.Warning)
+    }
+
+    if (-not $DryRun)
+    {
+        Write-Host $([Constants]::DoubleDashLine)
+        Write-Host "[Step 4 of 4] Set the Cluster Protection Level to EncryptAndSign on Service Fabric(s) in the Subscription..." 
+        Write-Host $([Constants]::SingleDashLine)
+        
+
+        if (-not $Force)
+        {
+            Write-Host "Do you want to set Cluster Protection level to EncryptAndSign on Service Fabric(s) in the Subscription? " -ForegroundColor $([Constants]::MessageType.Warning)
+            
+            $userInput = Read-Host -Prompt "(Y|N)"
+
+            if($userInput -ne "Y")
+            {
+                Write-Host "Cluster Protection level will not be changed to EncryptAndSign on Service Fabric(s) in the Subscription. Exiting..." -ForegroundColor $([Constants]::MessageType.Warning)
+                break
+            }
+        }
+        else
+        {
+            Write-Host "'Force' flag is provided. Cluster Protection level will be changed as EncryptAndSign on Service Fabric(s) in the Subscription without any further prompts." -ForegroundColor $([Constants]::MessageType.Warning)
+        }
+
+        # List for storing remediated Service Fabric(s)
+        $ServiceFabricRemediated = @()
+
+        # List for storing skipped Service Fabric(s)
+        $ServiceFabricSkipped = @()
+        $ServiceFabricSkippedWithSingleNode = @()
+
+        Write-Host "Setting ClusterProtectionLevel to EncryptAndSign on Service Fabric(s)." -ForegroundColor $([Constants]::MessageType.Info)
+
+        # Loop through the list of Service Fabric(s) which needs to be remediated.
+        $ServiceFabricWithoutEncryptandSign | ForEach-Object {
+            $ServiceFabric = $_
+            try
+            {
+                if($_.NodeVMCount -lt 3)
```

Comment:
Is this should be ($_.NodeVMCount -eq 1)

Reply:
Hi Tarun, its resolved and pushed.